src/main/java/trello/api/rest/RestClientAPI.java

```diff
@@ -12,57 +12,106 @@
+    }
+
+    /**
+     * Gets a response of type put.
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response put(final String endpoint) {
+        return apiResponse("PUT", endpoint);
+    }
 
-    private static RestClientAPI oauth;
-    private RequestSpecification request;
+    /**
+     * Gets a response of type post.
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response post(final String endpoint) {
+        return apiResponse("POST", endpoint);
+    }
 
     /**
-     * This is constructor that initializes variables.
+     * Gets a response of type delete.
+     * @param endpoint defines the type the http method.
+     * @return a response.
      */
-    private RestClientAPI() {
-        User user = JsonConverter.jsonToUser(ReadJsonFile.getInstance().getDataByUserType("admin"));
-        String consumerKey = user.getConsumerKey();
-        String consumerSecret = user.getConsumerSecret();
-        String accessToken = user.getAccessToken();
-        String tokenSecret = user.getTokenSecret();
-        String baseUrl = NamePages.getBaseUrlAPI();
-        request = new RequestSpecBuilder().setAuth(RestAssured.oauth(consumerKey, consumerSecret,
-                accessToken, tokenSecret)).setBaseUri(baseUrl).build();
+    public Response delete(final String endpoint) {
+        return apiResponse("DELETE", endpoint);
     }
 
     /**
-     * Gives the class instance according Singleton pattern.
-     *
-     * @return an instance.
+     * Gets a requestSpecification for RestClientAPI.
+     * @return a requestSpecification.
      */
-    public static RestClientAPI getInstance() {
-        if (oauth == null) {
-            oauth = new RestClientAPI();
-        }
-        return oauth;
+    public RequestSpecification getRequest() {
+        return requestSpecification;
     }
 
     /**
-     * Gives the request specification resultant of oauth.
-     *
-     * @return an request specification.
+     * Sets the attribute requestSpecification en RestClientAPI.
+     * @param requestSpecification defines a requestSpecification in attribute to be set.
      */
-    public RequestSpecification getRequestSpecification() {
-        return request;
+    public void setRequest(final RequestSpecification requestSpecification) {
+        this.requestSpecification = requestSpecification;
+    }
+
+    /**
+     * Builds the request defines the method http and endpoint.
+     * @param httpMethod defines the type the http method.
+     * @param endpoint defines the endpoint.
+     * @return a response.
+     */
+    private Response apiResponse(final String httpMethod, final String endpoint) {
+        return given().
+                spec(requestSpecification).
+                when().
+                request(httpMethod, endpoint);
+    }
+
+    /**
+     * Builds the request with a body.
```

Comment:
Please add a white space bellow this line.

Reply:
Solved